src/server.c

```diff
@@ -3710,6 +3710,60 @@ int commandCheckArity(client *c, sds *err) {
     return 1;
 }
 
+int findDuplicatesOptions(client *c, const char *token) {
+    int matched = 0;
+    for (int j = 2, k = c->argc - 1; j <= k; j++, k--) {
+        if (!strcasecmp(token, c->argv[j]->ptr)) {
+            matched++;
+        }
+        if (j != k && !strcasecmp(token, c->argv[k]->ptr)) {
+            matched++;
+        }
+        if (matched > 1) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+/* Check if c->cmd has duplicate options, fills `err` with details in case it has.
+ * Return 1 if no duplicate options are found. */
+int commandCheckArgTokenDuplicacy(client *c, sds *err) {
+    if (c->argc <= 4) return 1;
+    for (int iter = 0; iter < c->cmd->num_args; iter++) {
+        if (c->cmd->args[iter].token && !(c->cmd->args[iter].flags & CMD_ARG_MULTIPLE_TOKEN)) {
+            if (!findDuplicatesOptions(c, c->cmd->args[iter].token)) {
+                if (err) {
+                    *err = sdsnew(NULL);
+                    *err = sdscatprintf(*err, "duplicate '%s' option for '%s' command", c->cmd->args[iter].token,
+                                        c->cmd->fullname);
+                }
+                return 0;
+            }
+        }
+        if (c->cmd->args[iter].subargs) {
+            for (int subiter = 0; subiter < c->cmd->args[iter].num_args; subiter++) {
+                if (c->cmd->args[iter].subargs[subiter].token &&
+                    !(c->cmd->args[iter].subargs[subiter].flags & CMD_ARG_MULTIPLE_TOKEN)) {
+                    if (!findDuplicatesOptions(c, c->cmd->args[iter].subargs[subiter].token)) {
+                        if (err) {
+                            *err = sdsnew(NULL);
+                            *err = sdscatprintf(*err, "duplicate '%s' option for '%s' command",
+                                                c->cmd->args[iter].subargs[subiter].token, c->cmd->fullname);
+                        }
+                        return 0;
+                    }
```

Comment:
Currently, the implementation is designed to return after the first match. This approach is used because if a command has a large number of arguments, we would simply loop through to retrieve the entire list. I believe it would be more efficient to return immediately upon finding the first match.